src/elastio-snap.c

```diff
@@ -2203,6 +2211,8 @@ static int file_allocate(struct snap_device *dev, struct file *f, uint64_t offse
 	}
 
 out:
+	file_lock(f);
```

Comment:
It's correct to lock the cow file at the very end, after all zero fillings are made.